src/core/toga/widgets/optioncontainer.py

```diff
@@ -117,12 +117,12 @@ def _insert(self, index, label, widget, enabled=True):
         # Add the option to the list maintained on the interface,
         # and increment the index of all items after the one that was added.
         self._options.insert(index, option)
-        for option in self._options[index + 1:]:
-            option._index += 1
+        for opt in self._options[index + 1:]:
+            opt._index += 1
```

Comment:
`option` is already in use: assigned in the first line of the function, and used in the last. 
When the for loop runs (in the `insert` scenario), option would be assigned to the last tab, instead of the tab we are inserting.